### PR TITLE
Fix and clean up NuGet package publishing

### PR DESCRIPTION
**Pull request by tmat**
PR #29408 changed the layout of NuGet output directories but did not update publishing script and data accordingly.

### CODE REVIEW

**Comment by tmat** [comment]
@jaredpar @dotnet/roslyn-infrastructure PTAL

File: src/Tools/MicroBuild/publish-assets.ps1 [hunk 1]
```diff
@@ -96,9 +96,10 @@ function Publish-Channel([string]$packageDir, [string]$name) {
 function Test-Entry($publishData, [switch]$isBranch) { 
     if ($isBranch) { 
         if ($publishData.nuget -ne $null) { 
-            $kind = $publishData.nugetKind;
-            if ($kind -ne "PerBuildPreRelease") {
-                throw "Branches are only allowed to publish PerBuildPreRelease"
+            foreach ($nugetKind in $publishData.nugetKind) {
+                if ($kind -ne "PerBuildPreRelease" -and $kind -ne "Shipping" -and $kind -ne "NonShipping") {
```
[the remark this answers is not in the document]
Comment (tmat) [reply src/Tools/MicroBuild/publish-assets.ps1]: NonShipping/NonShipping is not about the version suffix.
It's about where we deploy the packages. Shipping means shipping on nuget.org. NonShipping means not shipping on nuget.org

[the remark this answers is not in the document]
Comment (tmat) [reply src/Tools/MicroBuild/publish-assets.ps1]: Here we want to publish all Shipping and NonShipping to myget.

**Comment by tmat** [comment]
test ubuntu_16_debug_prtest please

File: src/Tools/MicroBuild/publish-assets.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmarolf) [reply src/Tools/MicroBuild/publish-assets.ps1]: `$kind` needs to be renamed to `$nugetKind`

[the remark this answers is not in the document]
Comment (tmat) [reply src/Tools/MicroBuild/publish-assets.ps1]: @jmarolf Good catch.

**Pull request merged**